### PR TITLE
TIL: how to replace an external Go dependency with a local copy

### PR DESCRIPTION
**Pull request by zmoog**
### Motivation

I want to start publishing on my blog a series of small TIL posts and then over time move up to more long form content.

### Change description

Add a post on how to use `replace` directive to replace a dependency module with locla copy.

### Additional Notes

¯\_(ツ)_/¯

### CODE REVIEW

File: content/post/til-replace-module.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+draft: true
+---
+
+## The Problem
+
+You're using a great open source library in your project, and then, one day, you find yourself thinking: 
+
+> "Oh, wouldn't be great if the author just added a tiny little log statement in that function?". 
+
+Yeah, it really would, thiw would allow me to trace the value of that variable o try a small change.
```
Comment (xrmx) [inline content/post/til-replace-module.md]: Thiw -> this

Comment (xrmx) [reply content/post/til-replace-module.md]: O -> or (or to?)

File: content/post/til-replace-module.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    github.com/zmoog/bar v0.1.0
+)
+```
+
+
+## Conclusion
+
+That's it!
+
+Now, when you recompile or run your code, the local copy will be used with all your local changes.
```
Comment (xrmx) [inline content/post/til-replace-module.md]: You should add more emphasis that the change should be transient and never committed

Comment (zmoog) [reply content/post/til-replace-module.md]: Oh, you're right! I didn't think about this aspect, thank you!

**Review by xrmx — APPROVED** [review]
:tada:

File: content/post/til-replace-module.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+```
+module github.com/zmoog/foo
+
+require (
+    github.com/zmoog/bar v0.1.0
+)
+```
+
+To replace the module `github.com/zmoog/bar` with a local copy located on your local disk, all you need to do is add the a new line with the [replace](https://golang.org/ref/mod#go-mod-file-replace) directive:
```
Comment (pollon) [inline content/post/til-replace-module.md:35]: all you need to do is add **the** a new line -> all you need to do is add a new line

Comment (zmoog) [reply content/post/til-replace-module.md:35]: Thank you! ❤️  Fixed in #3.

**Pull request merged**